shell/app-shell/elements/cloud-data/cloud-handles.js

```diff
@@ -36,15 +36,29 @@ class CloudHandles extends Xen.Debug(Xen.Base, log) {
   }
   _scanHandles(key, arc, roots) {
     const paths = {};
-    [...arc._storeTags].forEach(tagEntry => this._handleToPath(paths, key, tagEntry));
+    [...arc._storeTags].forEach(tagEntry => this._handleToPaths(paths, key, tagEntry));
     this._updateWatches(arc, roots, paths);
   }
-  _handleToPath(paths, key, [localHandle, tags]) {
-    if (tags && tags.size > 0 && !tags.has('nosync')) {
-      const contextId = ArcsUtils.getContextHandleId(localHandle.type, tags);
-      const path = `arcs/${key}/${Const.DBLABELS.handles}/${contextId}`;
-      paths[path] = localHandle;
-    }
+  _handleToPaths(paths, key, [localHandle, tags]) {
+    const {type} = localHandle;
+    if (type.isCollection || type.isEntity) {
+      // TODO(sjmiles): the contextId has to be unique for this arc, but also persistent
```

Comment:
Is this something to discuss with runtime team as follow-up?

Reply:
I figured not, because this all goes away when we can use Runtime Storage. 

Of course, Storage is blocked on Serialization, which is why I'm a dog with that bone!